aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -111,8 +113,12 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     @classmethod
-    def define(cls, spec):
+    def define(cls, spec: CalcJobProcessSpec):
```

Comment:
So, this does not change behavior, it is just for documentation's purposes, right? If so, maybe we should do the same in `aiida/workflows/arithmetic/multiply_add.py`(line 30).

Reply:
yes, this is a feature called type hinting that was introduced in Python 3.5. You can directly specify the type of arguments and return values. This is not just for documentation, but also enables some form of static type analysis which can help IDE's and linting tools

Reply:
If we're going to start adding type hints, we should probably run `mypy` (with the least restrictive options, to start) in the pre-commit hooks. Having _wrong_ type hints is definitely worse than having none at all.

Reply:
Will give at try momentarily.

Reply:
``Found 664 errors in 240 files (checked 724 source files)`` ooh boy 🥴

Reply:
Huh? What are these errors? Are these type violations detected by `mypy`? How is that possible if we don't even specify types yet?

Reply:
So yeah, enabling it on the whole codebase at once may be a tad too ambitious. Here's a guide how to do it step-by-step: https://mypy.readthedocs.io/en/stable/existing_code.html

Reply:
It'll also detect type violations for things it can infer automatically.. like overwriting a variable with something of a different type. And there's also a bunch of errors from things it doesn't know how/where to import, because it hasn't been annotated yet.